docs/application/dotnet/guides/nui/visuals.md

```diff
@@ -1,638 +1,772 @@
+
+   ColorVisual _colorVisual_2 = new ColorVisual();
+
+   // Obligatory properties
+   _colorVisual_2.Color = new Color(0.0f, 0.0f, 1.0f, 0.8f);
+
+   // Optional properties:
+   _colorVisual_2.Size = new Size2D(300,300);
+   _colorVisual_2.Origin = Visual.AlignType.BottomEnd;
+   _colorVisual_2.AnchorPoint = Visual.AlignType.BottomEnd;
+   // Shifting the visual left and up 
+   _colorVisual_2.RelativePosition = new RelativeVector2(-0.1f, -0.1f);
+   // Setting a value lower than for the previous object causes it to be covered
+   // Without changing this value, the objects are drawn in the order they were added
+   _colorVisual_2.DepthIndex = _colorVisual_1.DepthIndex - 1;
+
+   _visualView.AddVisual("BlueRectangle", _colorVisual_2);
+   ```
+
+The final output after implementing the preceding code, and setting the `_visualView` background color to gray is as follows:
+
+ | `DepthIndex` default values.                         | `DepthIndex` changed values.                                         |
+ |------------------------------------------------------|----------------------------------------------------------------------|
+ | ![ColorVisual](media/visuals/ColorVisualExample.png) | ![ColorVisualDepth](media/visuals/ColorVisualExample_DephtIndex.png) |
+
+### GradientVisual
+
+The `GradientVisual` renders a smooth transition of colors. NUI supports both linear and radial gradients.
+
+The following table lists the supported properties:
+
+**Table: GradientVisual properties.**
+
+| Property        | Type                             | Required        | Description                                                                                    |
+|-----------------|----------------------------------|-----------------|------------------------------------------------------------------------------------------------|
+| `StartPosition` | `Vector2`                        | For linear only | The start position of the linear gradient. The coordinate system depends on the `Units` value. |
+| `EndPosition`   | `Vector2`                        | For linear only | The end position of the linear gradient. The coordinate system depends on the `Units` value.   |
+| `Center`        | `Vector2`                        | For radial only | The center point of the radial gradient. The coordinate system depends on the `Units` value.   |
+| `Radius`        | `float`                          | For radial only | The size of the radial gradient radius. The coordinate system depends on the `Units` value.    |
+| `StopColor`     | `PropertyArray` of `Color`       | Yes             | The color at the stop offsets. At least two are required to show a gradient.                   |
+| `StopOffset`    | `PropertyArray` of `floats`      | No              | The stop offsets in relative units. To see all the colors from the `StopColor`, the lengths of these two arrays should be the same. The default values are `0.0` and `1.0`. |
```

Comment:
For more clarity
```suggestion
| `StopOffset`    | `PropertyArray` of `floats`      | No              | The stop offsets in relative units. To see all the colors from the `StopColor`, the length of the `PropertyArray` of `floats`  should be the same. The default values are `0.0` and `1.0`. |
```
Please check if this is technically correct; if no, request you to correct the same.

Reply:
no, but I changed it. please check it now